### ISSUE TITLE
[Bug Report][3.10.10] [v-dialog] Focus trap fails (focus leaves dialog) after closing a nested dialog and pressing Tab

### ISSUE THREAD

**Issue by NhuHoang28** [vuetifyjs/vuetify#22322]
### Environment
**Vuetify Version:** 3.10.10
**Vue Version:** 3.5.24
**Browsers:** Chrome 142.0.7444.60
**OS:** Windows 11

### Steps to reproduce
1. Click button "Open Dialog 1"
2. Click button "Open Dialog 2"
3. Click button "Close" on Dialog 2
4. Press Tab key

### Expected Behavior
Focus remains on the elements inside the dialog

### Actual Behavior
Focuses outside the dialog, on background element

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNrdVV1PwjAU/Su1L2rCQAETYybR6LuJr+JD7Yo2dmvTXhBD+O/ernTUCYEXP+LCw93u6dk9p4f1YUGd5b1rY7qzqaAXNAdRGsVAjMYVIXkhZ4Qr5tzlmBqWDQmIOWRcVCDsmNYYRM2yJ6hCTWoEou+MqMitZEo/k9Mxjd0rriR/xX4RWpcE7FTE/ijv1VxI3FAHYFw/y0pdCNUQrJlLNs/eZAEv2BuenzSUsY9UnNmCgAQlELOerYEgKMrHFzml4cKrSfptseGKDpXvWX890FY7vmBapvRbroQr8aaZpZfsVvI4KM0YB6kr1x7fGcaFrelWZbp4s8KVihulXWss3BJmJat826N2aENpE6baJBu1bVERGzF8eBuo9w9NsgFpavrDvVODDD+dmsEOZwdo7cGq/I/J8f+K349OYm2MDZuC3jc3uDwd9A/b7eP0jXb7WzxcsGpFMXfcSgPECZiaGihLoy2QBXEvTCn9di8mZEkmVpfkEA+tw2AKxzcDaT4ya+xRreK4jfFx2gnyJmwA5b0w5IguO3TQPev2h9QXpyf4o50a1vl0sIZnjx/V1yI/)

### Other comments
**The same behavior occurs when:**

Case 1:
1. Click button "Open Dialog 1"
2. Click on tab "PREVIEW" of Vuetify Playground
3. Press Tab key

Case 2:
1. Click button "Open Dialog 1"
2. Press "Shift + Tab" key

**Comment by KaelWD** [comment]
> Click button "Close" on Dialog 2
> Press Tab key

This works correctly in firefox but not chrome.

> Click on tab "PREVIEW" of Vuetify Playground
> Press Tab key

Expected, the playground uses an iframe so we can't intercept tab presses from outside. Click-outside also doesn't work with iframes.

> Press "Shift + Tab" key

Looks like this was broken by #22101.

**Comment by NhuHoang28** [comment]
> Click on tab "PREVIEW" of Vuetify Playground
> Press Tab key

> Expected, the playground uses an iframe so we can't intercept tab presses from outside. Click-outside also doesn't work with iframes.

Thanks for the note on iFrame limitations.
However, this is a regression starting from v3.10.4. The behavior described (Focus remains in dialog) worked in version 3.10.3 and earlier, even within the Playground iFrame.

Also, here is a non-iFrame case that fails, steps to reproduce:
1. Click button "Open Dialog 1"
2. Click on browser URL bar
3. Press Tab key repeatedly

Expected: 
Focus returns to and remains inside the dialog.

Actual (v3.10.4+): 
Focus lands on a background element outside the dialog.

### Video demo:
**On v3.10.3:** https://github.com/user-attachments/assets/568d2a8e-071d-4ecd-9b1c-0539326e5873
**On v3.10.4:** https://github.com/user-attachments/assets/541f078a-9ae1-478f-b574-49f27faa1c0d
